compiler/rustc_next_trait_solver/src/solve/trait_goals.rs

```diff
@@ -228,37 +229,17 @@ where
         ecx.probe_and_evaluate_goal_for_constituent_tys(
             CandidateSource::BuiltinImpl(BuiltinImplSource::Trivial),
             goal,
-            structural_traits::instantiate_constituent_tys_for_sized_trait,
-        )
-    }
-
-    fn consider_builtin_metasized_candidate(
-        ecx: &mut EvalCtxt<'_, D>,
-        goal: Goal<I, Self>,
-    ) -> Result<Candidate<I>, NoSolution> {
-        if goal.predicate.polarity != ty::PredicatePolarity::Positive {
-            return Err(NoSolution);
-        }
-
-        ecx.probe_and_evaluate_goal_for_constituent_tys(
-            CandidateSource::BuiltinImpl(BuiltinImplSource::Misc),
-            goal,
-            structural_traits::instantiate_constituent_tys_for_sized_trait,
-        )
-    }
-
-    fn consider_builtin_pointeesized_candidate(
-        ecx: &mut EvalCtxt<'_, D>,
-        goal: Goal<I, Self>,
-    ) -> Result<Candidate<I>, NoSolution> {
-        if goal.predicate.polarity != ty::PredicatePolarity::Positive {
-            return Err(NoSolution);
-        }
-
-        ecx.probe_and_evaluate_goal_for_constituent_tys(
-            CandidateSource::BuiltinImpl(BuiltinImplSource::Misc),
-            goal,
-            structural_traits::instantiate_constituent_tys_for_sized_trait,
+            match sizedness {
+                SizedTraitKind::Sized => {
+                    structural_traits::instantiate_constituent_tys_for_sized_trait
+                }
+                SizedTraitKind::MetaSized => {
+                    structural_traits::instantiate_constituent_tys_for_metasized_trait
+                }
+                SizedTraitKind::PointeeSized => {
+                    structural_traits::instantiate_constituent_tys_for_pointeesized_trait
+                }
+            },
```

Comment:
I don't think we this extra layer of indirection as there is already `instantiate_constituent_tys_for_sizedness_trait` which takes any `SizedTraitKind`

Reply:
Fixed